[Tammy Koh] iP

tammykoh opened the pull request.

tammykoh committed:
Duke greets the user

tammykoh committed:
Include function where Duke echos the commands entered by user, and exits when the user types bye

tammykoh committed:
Duke is now able to add items to a list, and display them when requested

tammykoh committed:
Create Task Class, and Duke is now able to mark task as done

tammykoh committed:
Tidy code and ensure it complies to coding standard

yuen-sihao commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,92 @@
+
+    // List of all tasks
+    private static Task[] Tasks = new Task[MAX_TASK];
+
```
I like how you used a variable name for MAX_TASK instead of "magic numbers".

yuen-sihao reviewed (commented):
Overall, I think the code looks good based on coding standards.

GuoAi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,92 @@
+
 public class Duke {
+
+     // Maximum number of tasks that can be held
+    private static final int MAX_TASK = 100;
```
I like how you defined a constant variable instead of using magic literals.

lowjiayee replied:
MAX_TASK is self-explanatory, you may consider removing the comment.

GuoAi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,92 @@
+            String userCommand = getUserInput();
+            executeUserCommand(userCommand);
+        }
+    }
```
I like how you followed "SLAP" (single level of abstraction per function) in your main function. All the methods in your main function have the same level of abstraction.

GuoAi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,92 @@
+        default:
+            echoUserCommand(rawUserCommand);
+            break;
+        }
```
I like how you used switch here and that you had "break" after each case statement.

GuoAi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,92 @@
+        }
+    }
+
+    private static void echoUserCommand(String userCommand) {
```
I like how you made your method private to prevent other classes from calling it.

GuoAi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,92 @@
+     // Maximum number of tasks that can be held
+    private static final int MAX_TASK = 100;
+
+    // List of all tasks
+    private static Task[] Tasks = new Task[MAX_TASK];
```
Should variables be named using camelCase?

lowjiayee replied:
Perhaps using "taskList" or "tasks" would represent this better? If you use "taskList", I would think that its purpose will be clear enough such that the comment on line 8 becomes unnecessary.

GuoAi reviewed (commented):
Overall, I find your code very readable and clean. I learned from your code as well. Good job!

yuen-sihao commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,92 @@
+    private static void displayList() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i=0; i<count; i++) {
+            int taskNumber = i + 1;
```
Consider having spaces before and after the operator signs like '=' and '<' to make it more readable.

yuen-sihao commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,18 @@
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
```
Consider leaving out the "this" word when they are not necessary.

yuen-sihao reviewed (commented):
I like how all your methods have names that tells the user what it does. Overall, still considered a well written code, just giving another look to your code and found possibly more learning points.

lowjiayee commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,18 @@
+
+    public String getStatusIcon() {
+        //return tick or X symbols
+        return (isDone ? "[\u2713]" : "[\u2718]");
```
Perhaps you can use a constant to illustrate this unicode better.

lowjiayee commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,92 @@
+            displayList();
+            break;
+        case "done":
+            int taskNumber = Integer.parseInt(rawUserCommand.substring(dividerPosition+1));
```
I would suggest to separate substring() and parseInt() into 2 different lines to improve readability.

tammykoh committed:
Add ToDo, Event and Deadline class to support each type of task respectively

tammykoh committed:
Semi-automated the testing of Duke, and examined code to improve code quality

tammykoh committed:
Duke is now able to handle some errors
Created DukeException class to handle errors specific to duke

tammykoh committed:
Organise class into packages

tammykoh committed:
Merge branch 'branch-Level-5'

tammykoh committed:
Duke is now able to delete tasks. Tasks are now stored using ArrayList.

tammykoh committed:
Duke saves the task list in "./data/tasks.txt" whenever the task list changes
In the event the folder and/or file does not exist, Duke will create it for the user

tammykoh committed:
Merge branch 'branch-Level-6'

tammykoh committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/duke/Duke.java

tammykoh committed:
Created JAR file

tammykoh committed:
Refactored the code to extract closely related code as classes

tammykoh committed:
Users can now find their task by searching for a keyword

tammykoh committed:
Added comments to the code

tammykoh committed:
Merge pull request #1 from tammykoh/branch-Level-9

Merge branch-Level-9

tammykoh committed:
Merge pull request #2 from tammykoh/branch-A-JavaDoc

Merge branch-A-JavaDoc

tammykoh committed:
Added a user guide on how to use Duke

tammykoh committed:
Duke can handle more errors
Customised Duke

tammykoh committed:
Fixed bug in adding task to file and loading task file

tammykoh committed:
Fixed minor bugs and cleaned up the code

tammykoh committed:
Updated the User Guide and moved it to the right folder

tammykoh committed:
Updated link to download JAR file

tammykoh committed:
Set theme jekyll-theme-time-machine

tammykoh committed:
Set theme jekyll-theme-tactile

tammykoh committed:
Delete _config.yml

Removed theme

tammykoh committed:
Improved comments and code quality

tammykoh committed:
Merge branch 'master' of https://github.com/tammykoh/ip
